close rsvp and cfp form when event ends

idlip opened the pull request:
- helps to hide forms and avoid manual closure by team

closes #1096

Unchecks (close) both RSVP and CFP form after event ends.

<!-- This is an auto-generated comment: release notes by coderabbit.ai -->
## Summary by CodeRabbit

* **Bug Fixes**
  * Scheduled task now processes both Live and Cancelled events, marking them Concluded or Cancelled, clearing RSVP/CFP visibility, and updating linked RSVP forms to be unpublished and CFPs to be closed.
* **Tests**
  * Added tests and helpers validating event conclusion behavior and RSVP/CFP state transitions across Live, Cancelled, and Concluded scenarios (with controlled current-date patching).
<!-- end of auto-generated comment: release notes by coderabbit.ai -->

idlip committed:
close rsvp and cfp form when event ends

- helps to hide forms and avoid manual closure by team

rahulporuri commented on fossunited/scheduled_tasks.py:
```diff
@@ -19,6 +19,8 @@ def conclude_events():
         doc = frappe.get_doc(EVENT, event.name)
         doc.status = "Concluded"
+        doc.show_rsvp = 0
+        doc.show_cfp = 0
```
this will simply hide the rsvp and cfp forms linked with the event - but not "unpublish" or "close" them

for that, we will need to pull the associated CFP and RSVP doctypes and update the `status` on CFP to `Closed` and `is_published` on RSVP to `0`. Ref `FOSSEventCFP` and `FOSSEventRSVP` doctypes

idlip replied:
Ohh point..
Sorry for missing the flow here.

Will update!

idlip replied:
Pushed the fix.

idlip committed:
make RSVP+CFP get closes and unpublish them

idlip commented:
@rahulporuri Pushed the fix.

Now it fetches RSVP and CFP doc info for and sets CFP status to closed and rsvp to be unpublished.

idlip commented:
Hold! I will investigate unit case failing!

idlip committed:
add unit test for proper checks

idlip commented:
Latest change:

- Added unit tests

Made concluding event also check for cancelled and close RSVP and CFP is they are open (just to be safe)

- Added unit test to include both forms based on 3 events
  - event1 was live and ended so test checks to be concluded and if both forms are closed.
  - event2 was live but cancelled, so again we expect same to be closed
  - event3 is live and both should be open/published.

Unit test pass. The scheduler should close event, and both forms as expected.

rahulporuri reviewed (approved):
this looks good - lets merge ~and check tomorrow to see if some of the open RSVPs/CFPs got closed.~

sorry - I think you'll need to manually close/unpublish some of the current RSVPs/CFPs because this script won't close them

rahulporuri commented on fossunited/scheduled_tasks.py:
```diff
@@ -11,16 +18,30 @@ def conclude_events():
-        doc.status = "Concluded"
+        if doc.status == "Live":
+            doc.status = "Concluded"
+        else:
+            doc.status = "Cancelled"
```
why do we need this else?

idlip replied:
From doctypes, we grab all events that are ended but are [Live & Cancelled]
Actually we close only for live events.

I added `else` for cancelled events cause there might be any RSVP or CFP forms still open.
Although ideally I dont think cancelled would have them open. But just a safer check to turn them if found.

Only special case this applies is, an event is made which ends next week with both forms open/live, then it got cancelled due to some reason. So this scheduler should act as safer check to close them if so found

idlip committed:
fix handling of form fix not found

idlip commented on fossunited/scheduled_tasks.py:
```diff
@@ -11,16 +18,30 @@ def conclude_events():
             doc.save(ignore_permissions=True)
+            past_rsvp.save(ignore_permissions=True)
+            past_cfp.save(ignore_permissions=True)
         except Exception as e:
```
makes sense. Added it!

idlip commented:
hopefully it should work.

Logic seems fine, if I missed something, we will know ahead :)
